internal/state/change_processor_test.go

```diff
@@ -278,988 +244,190 @@ var _ = Describe("ChangeProcessor", func() {
-									{
-										Path:     "/",
-										PathType: dataplane.PathTypePrefix,
-										MatchRules: []dataplane.MatchRule{
-											{
-												MatchIdx:     0,
-												RuleIdx:      0,
-												BackendGroup: hr1Group,
-												Source:       hr1Updated,
-											},
-										},
-									},
-								},
-							},
-						},
-						SSLServers: []dataplane.VirtualServer{
-							{
-								IsDefault: true,
-							},
-							{
-								Hostname: "foo.example.com",
-								SSL:      &dataplane.SSL{CertificatePath: certificatePath},
-								PathRules: []dataplane.PathRule{
-									{
-										Path:     "/",
-										PathType: dataplane.PathTypePrefix,
-										MatchRules: []dataplane.MatchRule{
-											{
-												MatchIdx:     0,
-												RuleIdx:      0,
-												BackendGroup: hr1Group,
-												Source:       hr1Updated,
-											},
-										},
-									},
-								},
-							},
-							{
-								Hostname: "~^",
-								SSL:      &dataplane.SSL{CertificatePath: certificatePath},
-							},
-						},
-						BackendGroups: []dataplane.BackendGroup{hr1Group},
-					}
-					expectedStatuses := state.Statuses{
-						GatewayClassStatus: &state.GatewayClassStatus{
-							ObservedGeneration: gc.Generation,
-							Conditions:         conditions.NewDefaultGatewayClassConditions(),
-						},
-						GatewayStatuses: state.GatewayStatuses{
-							{Namespace: "test", Name: "gateway-1"}: {
-								Conditions:         conditions.NewDefaultGatewayConditions(),
-								ObservedGeneration: gw1Updated.Generation,
-								ListenerStatuses: map[string]state.ListenerStatus{
-									"listener-80-1": {
-										AttachedRoutes: 1,
-										Conditions:     conditions.NewDefaultListenerConditions(),
-									},
-									"listener-443-1": {
-										AttachedRoutes: 1,
-										Conditions:     conditions.NewDefaultListenerConditions(),
-									},
-								},
-							},
-						},
-						HTTPRouteStatuses: state.HTTPRouteStatuses{
-							{Namespace: "test", Name: "hr-1"}: {
-								ObservedGeneration: hr1Updated.Generation,
-								ParentStatuses: []state.ParentStatus{
-									{
-										GatewayNsName: client.ObjectKeyFromObject(gw1Updated),
-										SectionName:   helpers.GetPointer[v1beta1.SectionName]("listener-80-1"),
-										Conditions:    conditions.NewDefaultRouteConditions(),
-									},
-									{
-										GatewayNsName: client.ObjectKeyFromObject(gw1Updated),
-										SectionName:   helpers.GetPointer[v1beta1.SectionName]("listener-443-1"),
-										Conditions:    conditions.NewDefaultRouteConditions(),
-									},
-								},
-							},
-						},
-					}
-
-					changed, conf, statuses := processor.Process(context.TODO())
+					changed, graphCfg := processor.Process()
 					Expect(changed).To(BeTrue())
-					Expect(helpers.Diff(expectedConf, conf)).To(BeEmpty())
-					assertStatuses(expectedStatuses, statuses)
+					Expect(graphCfg.Gateway).ToNot(BeNil())
```

Comment:
Should we also check routes and gateway class here? If I'm reading these tests correctly, there should still be a gateway class and 1 route in the cluster state.

Reply:
Yep, missed that one.